assessments/projects/longhorn/self-assessment.md

```diff
@@ -0,0 +1,419 @@
+
+#### 2.1 Components
+- **Longhorn Manager**
+- **Longhorn Engine**
+- **Replicas**
+- **Nodes and Disks**
+
+#### 2.2 Actors
+- **Longhorn Manager**
+- **Longhorn Engine**
+- **Users/Administrators**
+
+#### 2.3 Actions
+- **Volume Provisioning and Attachment**
+- **Data Replication**
+- **Snapshot and Backup Operations**
+- **Volume Restoration and Recovery**
+
+### 3. STRIDE Analysis
+
+#### 3.1 Spoofing
+**Threats**:
+- Unauthorised access to the Longhorn management interface or API.
+- Impersonation of Longhorn components or services.
+
+**Mitigations**:
+- Implement strong authentication mechanisms for API and management interface access.
+- Use mutual TLS (mTLS) for internal communications.
+
+#### 3.2 Tampering
+**Threats**:
+- Unauthorised modifications to data in transit or at rest.
+- Tampering with Longhorn configuration or codebase.
+
+**Mitigations**:
+- Enable data encryption at rest and in transit.
+- Ensure data integrity through checksums and replication verification.
+- Employ strict access controls and code signing for codebase and configurations.
+
+#### 3.3 Repudiation
+**Threats**:
+- Denial of actions performed by users or internal processes.
+- Lack of auditing trails for critical operations.
+
+**Mitigations**:
+- Implement comprehensive logging and auditing mechanisms.
+- Ensure that all critical actions are traceable to specific users or entities.
+
+#### 3.4 Information Disclosure
+**Threats**:
+- Unauthorised access to sensitive data stored in Longhorn volumes.
+- Exposure of internal configuration or metadata.
+
+**Mitigations**:
+- Enforce strict access controls to data volumes.
+- Use encryption to protect sensitive data.
+- Restrict access to internal metadata and configuration details.
+
+#### 3.5 Denial of Service (DoS)
+**Threats**:
+- Overloading the Longhorn system, leading to unavailability.
+- Exploiting vulnerabilities to crash the system or degrade performance.
+
+**Mitigations**:
+- Implement rate limiting and access controls.
+- Design for high availability and resilience.
+- Regularly update and patch to address known vulnerabilities.
+
+#### 3.6 Elevation of Privilege
+**Threats**:
+- Exploitation of vulnerabilities to gain higher privileges.
+- Unauthorised access leading to control over Longhorn operations.
+
+**Mitigations**:
+- Adhere to the principle of least privilege in access controls.
+- Regular security assessments and penetration testing.
+- Monitor and promptly update software components to address vulnerabilities.
+
+### 4. Conclusion
+This STRIDE threat model for Longhorn identifies key areas of potential security risks and provides a foundation for implementing effective security measures. Regular updates, vigilant monitoring, and adherence to security best practices are essential to maintain the security and integrity of the Longhorn system.
+
+</details>
+
+
+<details>
+ 
+<summary>Longhorn Lightweight Threat Model</summary>
```

Comment:
@ragashreeshekar Is this supposed to be inlined here or moved to a different document?

Reply:
It is also mentioned in the document a few places.  This may be fine but would need to be patched up if it is externalized.

Reply:
Updated the document and its references